arcgis-ios-sdk-samples/Utility network/Display content of utility network container/DisplayContentUtilityNetworkContainerViewController.swift

```diff
@@ -0,0 +1,292 @@
+    /// The map view managed by the view controller.
+    @IBOutlet var mapView: AGSMapView! {
+        didSet {
+            mapView.map = makeMap()
+        }
+    }
+    /// The bar button item to prompt return of the main view.
+    @IBOutlet var exitBarButtonItem: UIBarButtonItem!
+    @IBOutlet var legendBarButtonItem: UIBarButtonItem!
+    
+    /// A feature service for an electric utility network in Naperville, Illinois.
+    let featureServiceURL = URL(string: "https://sampleserver7.arcgisonline.com/arcgis/rest/services/UtilityNetwork/NapervilleElectric/FeatureServer")!
+    var utilityNetwork: AGSUtilityNetwork?
+    let graphicsOverlay = AGSGraphicsOverlay()
+    /// The default or previous viewpoint before entering the container view.
+    var previousViewpoint: AGSViewpoint?
+    /// An array containing information about the feature services' legends.
+    var legendInfos = [AGSLegendInfo]()
+    var featureLayers = [AGSFeatureLayer]()
+    
+    // The symbols used to display the container view contents.
+    let boundingBoxSymbol = AGSSimpleLineSymbol(style: .dash, color: .yellow, width: 3)
+    let attachmentSymbol = AGSSimpleLineSymbol(style: .dot, color: .green, width: 3)
+    let connectivitySymbol = AGSSimpleLineSymbol(style: .dot, color: .red, width: 3)
+    
+    /// The action that is prompted when exiting the container view.
+    @IBAction func exitContainerView() {
+        // Disable the bar button item since container view will be exited.
+        exitBarButtonItem.isEnabled = false
+        // Remove all the objects that were added onto the graphics overlay.
+        graphicsOverlay.graphics.removeAllObjects()
+        (mapView.map?.operationalLayers as? [AGSLayer])?.forEach { layer in
+            // Make each operational layer visible.
+            layer.isVisible = true
+            guard let previousViewpoint = previousViewpoint else { return }
+            // Return to the viewpoint before container view was entered.
+            mapView.setViewpointGeometry(previousViewpoint.targetGeometry) { [weak self] _ in
+                // Enable interaction on the map view.
+                self?.mapView.isUserInteractionEnabled = true
+            }
+        }
+    }
+    
+    /// Create a map.
+    ///
+    /// - Returns: An `AGSMap` object.
+    func makeMap() -> AGSMap {
+        let webMapURL = URL(string: "https://ss7portal.arcgisonline.com/arcgis/home/item.html?id=5b64cf7a89ca4f98b5ed3da545d334ef")!
+        let map = AGSMap(url: webMapURL)!
+        return map
+    }
+    
+    /// Create and load the utility network using the feature service URL.
+    func loadUtilityNetwork() {
+        utilityNetwork = AGSUtilityNetwork(url: featureServiceURL, map: mapView.map!)
+        utilityNetwork?.load { [weak self] error in
+            // Add self as the touch delegate for the map view.
+            self?.mapView.touchDelegate = self
```

Comment:
I didn't put it in the else condition since someone pointed out that we should avoid pushing to the right due to nesting conditions. Not sure which is preferred

Reply:
Well I mean if there is an error when loading the UN, we probably shouldn't even allow the map view to detect tap events, to avoid unexpected outcomes.